glommio/src/sys/uring.rs

```diff
@@ -1604,14 +1601,16 @@ impl Reactor {
             membarrier::heavy();
             let events = process_remote_channels() + self.flush_syscall_thread();
             if events == 0 {
-                self.link_rings_and_sleep(&mut main_ring, &self.eventfd_src)
-                    .expect("some error");
+                if self.eventfd_src.is_installed().unwrap() {
```

Comment:
I wonder if we shouldn't register two eventfds for this? 

The wakeup code is tricky, and you are effectively adding one more condition.